File: Backend/Device Management System Backend/Device Management System Backend/Models/Device.cs
using Device_Management_System_Backend.Models.Enums;

namespace Device_Management_System_Backend.Models
{
    public class Device
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Manufacturer { get; set; } = string.Empty;
        public DeviceType Type { get; set; }
        public DeviceOperatingSystem OS { get; set; }
        public string OSVersion { get; set; } = string.Empty;
        public string Processor { get; set; } = string.Empty;
        public string RamAmount { get; set; } = string.Empty;
        public string? Description { get; set; }

        // FK
        public Guid? AssignedUserId { get; set; }
        // Navigation property
        public User? AssignedUser { get; set; }
    }
}

File: Frontend/device-management-frontend/src/app/models/device.ts
import { DeviceType, DeviceOperatingSystem } from './enums';

export interface Device {
    id: string; 
    name: string;
    manufacturer: string;
    type: DeviceType;
    os: DeviceOperatingSystem;
    osVersion: string;
    processor: string;
    ramAmount: string;
    description: string | null;
    assignedUserId: string | null; 
}